pinot-core/src/main/java/org/apache/pinot/server/access/ZkBasicAuthAccessFactory.java

```diff
@@ -97,7 +97,8 @@ public boolean hasDataAccess(RequesterIdentity requesterIdentity, String tableNa
             Map<String, ZkBasicAuthPrincipal> password2principal = name2password.keySet().stream()
               .collect(Collectors.toMap(name2password::get, _name2principal::get));
             return password2principal.entrySet().stream()
-                .filter(entry -> BcryptUtils.checkpw(entry.getKey(), entry.getValue().getPassword()))
+                .filter(entry -> BcryptUtils.checkpwWithCache(entry.getKey(), entry.getValue().getPassword(),
```

Comment:
name it consistent with the broker one. 
also why do we need 3? can't we just keep one in core?

Reply:
Thx for review, Do you means we can merge 
  org.apache.pinot.controller.api.access.ZkBasicAuthAccessControlFactory
  org.apache.pinot.broker.broker.ZkBasicAuthAccessControlFactory
to 
  org.apache.pinot.server.access.ZkBasicAuthAccessFactory as hole project ZkBasicAuthAccessFactory

Reply:
yes unless there's a specific reason controll/broker/server needs 3 different auth factory (and cannot be achieved via config differences)

Reply:
Thx @walterddr   But if we need to merge controller, broker, server  ZkBasicAuthAccessFactory,

Should we merge user and password for controller, broker, server nodes as well?  
In other words controller, broker and server nodes should use same user and password for hole pinot cluster